[#230] Add Bitrise template document (#283)
nimblehq/ios-templates: .github/wiki/Bitrise.md

| Bitrise.io Cache:Pull     | Bitrise.io Cache:Pull                                   | Bitrise.io Cache:Pull                   | Bitrise.io Cache:Pull                     |
| Run CocoaPods install     | Run CocoaPods install                                   | Run CocoaPods install                   | Run CocoaPods install                     |
| Fastlane - Build and Test | Xcode Test for iOS                                      | Xcode Test for iOS                      | Xcode Test for iOS                        |
| Fastlane - Run XCov       | Fastlane Match                                          | Fastlane Match                          | Fastlane Match                            |
| Danger                    | Fastlane - Build and Upload Production App to App Store | Fastlane - Build and Upload Staging App | Fastlane: Build and Upload Production App |

## Trigger Map

| Workflow                | Trigger                 |
|-------------------------|-------------------------|
| test                    | Create or Update a PR   |
| deploy_staging          | Push branch `develop`   |
| deploy_release_firebase | Push branch `release/*` |

> | deploy_app_store        | Push branch `master`    |

Reviewer: I think we can add the `main` branch also'
```suggestion
| deploy_app_store        | Push branch `master`/`main`  |
```

Author: Currently, in `bitrise.yml`, we don't have the trigger for the `main` branch. I think it is better to create a new issue to add a trigger for `main` as well as clean up unused env variables then I will update the document again.

## Environment and Secrets
### App Environtment Variables
- BITRISE_PROJECT_PATH
> e.g., ExampleApp.xcodeproj or in case you're using CocoaPod, it is ExampleApp.xcworkspace.

- TEAM_ID
> This is your Apple Team ID (e.g., T3T4E84BAA), you can find it in `Membership` at Apple developer portal.

- MATCH_REPO_URL
> Link to a repository that contains your Fastlane Match it can be either HTTPS or SSH link (e.g., https://github.com/nimblehq/fastlane-match.git)

### Workflow Environment Variables
All four workflows have their own following variables:

- BUNDLE_ID
> e.g., com.nimblehq.exampleApp

*Depending on which workflow, the value of those variables may differ from other workflows.*

### Secrets

- MATCH_PASSWORD
> This is an encryption password for the Match Repo

## Installation

> 1. Following the setup instruction in `README.md`.

Reviewer 2: What do you think about adding the link to `README.md`? 🙏

2. To connect your repository to Bitrise please following the instruction in this page: [Adding a new app](https://devcenter.bitrise.io/en/getting-started/adding-your-first-app.html).
3. Make sure the option where the `bitrise.yml` locate is set to `Store in-app repository`.

> ![bitrise.yml location](assets/images/bitrise/Bitrise-YML-Storage-Location.png)

Reviewer: We can improve this one to have a center and small image, what do you think?
```suggestion
<p align="center">
  <img src="assets/images/bitrise/Bitrise-YML-Storage-Location.png" alt="Bitrise Store in-app repository" width="600"/>
</p>
```

|-------------------------|-------------------------|
| test                    | Create or Update a PR   |
| deploy_staging          | Push branch `develop`   |
| deploy_release_firebase | Push branch `release/*` |
| deploy_app_store        | Push branch `master`    |

## Environment and Secrets
### App Environtment Variables
- BITRISE_PROJECT_PATH
> e.g., ExampleApp.xcodeproj or in case you're using CocoaPod, it is ExampleApp.xcworkspace.

- TEAM_ID
> This is your Apple Team ID (e.g., T3T4E84BAA), you can find it in `Membership` at Apple developer portal.

- MATCH_REPO_URL
> Link to a repository that contains your Fastlane Match it can be either HTTPS or SSH link (e.g., https://github.com/nimblehq/fastlane-match.git)

### Workflow Environment Variables
All four workflows have their own following variables:

- BUNDLE_ID
> e.g., com.nimblehq.exampleApp

*Depending on which workflow, the value of those variables may differ from other workflows.*

### Secrets

- MATCH_PASSWORD
> This is an encryption password for the Match Repo

## Installation
1. Following the setup instruction in [`README.md`](https://github.com/nimblehq/ios-templates#readme).

> 2. To connect your repository to Bitrise please following the instruction in this page: [Adding a new app](https://devcenter.bitrise.io/en/getting-started/adding-your-first-app.html).

Reviewer 3: ```suggestion
2. To connect your repository to Bitrise please follow the instruction on this page: [Adding a new app](https://devcenter.bitrise.io/en/getting-started/adding-your-first-app.html).
```
